docs/TensorDenotation.md

```diff
@@ -0,0 +1,48 @@
+# Tensor Denotation
+
+Tensor Denotation is an experimental attempt along with [Dimension Denotation](DimensionDenotation.md) to give tensor semantic description and enable model authors to describe what the input and output tensors are.
+
+## Motivation
+
+The motivation of such a mechanism can be illustrated via a simple example. In the the neural network SqueezeNet, it takes in an NCHW image input float[1,2,244,244] and produces a output float[1,1000,1,1]:
+
+```
+input_in_NCHW -> data_0 -> SqueezeNet() -> output_softmaxout_1
+```
+
+In order to run this model the user needs a lot of information.    In this case the user needs to know:
+* the input is an image
+* the image is in the format of NCHW
+* the color channels are in the order of bgr
+* the pixel data is 8 bit
+* the pixel data is normalized as values 0-255
+
+This proposal consists of three key components to provide all of this information: Tensor Denotation, [Dimension Denotation](DimensionDenotation.md), and [model metadata](MetadataProps.md).  Each of which will be discussed in detail.
+
+## Tensor Denotatio Definition
```

Comment:
n is missing -> Denotation

Reply:
good catch.   fixed in the the next iteration.